src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/DisableAzureDiskEncryption.cs

```diff
@@ -143,6 +143,34 @@ private VirtualMachineExtension GetVmExtensionParameters(VirtualMachine vmParame
                         null));
             }
 
+            string installedTypeHandlerVersion = null;
+            if (vmParameters.Resources != null)
+            {
+                foreach (VirtualMachineExtension vme in vmParameters.Resources)
+                {
+                    if (vme.Publisher.Equals(AzureDiskEncryptionExtensionContext.ExtensionDefaultPublisher) &&
+                         (vme.VirtualMachineExtensionType.Equals(AzureDiskEncryptionExtensionContext.ExtensionDefaultType) ||
+                          vme.VirtualMachineExtensionType.Equals(AzureDiskEncryptionExtensionContext.LinuxExtensionDefaultType)))
+                    {
+                        installedTypeHandlerVersion = vme.TypeHandlerVersion;
+                        break;
+                    }
+                }
+            }
+
+            if (installedTypeHandlerVersion == null)
+            {
+                ThrowTerminatingError(
```

Comment:
This adds a new failure condition where user won't be able to run 'Disable-AzureRmVmDiskEncryptionExtension' cmdlet on a VM that wasn't encrypted before . Or on a VM that was encrypted before but extension was removed (with encryption enabled).

Reply:
Thanks, good catch.  Didn't realize this was a supported use case, will revert this portion of the logic to previous behavior.

Reply:
@SudhakaraReddyEvuri added support for this case, please review new changes.